runtime/lua/vim/treesitter.lua

```diff
@@ -113,18 +113,26 @@ end
 ---
 --- If needed, this will create the parser.
 ---
+--- NOTE: This function will return `nil` rather than throwing an error in
+--- Nvim 0.12. Please pass `opts.error = false` to achieve this behavior
```

Comment:
a way to give ourselves a reminder about this in the version is to add a `test/functional/lua/deprecated_spec.lua` file with an `it()` test case that checks `vim.version.ge(vim.version(), '0.11')`